gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveLocationDescriptor.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.hive;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.Set;
+
+import lombok.Data;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.mapred.InputFormat;
+
+import gobblin.data.management.copy.RecursivePathFinder;
+import gobblin.util.PathUtils;
+
+/**
+ * Contains data for a Hive location as well as additional data if {@link #ENABLE_HIVE_LOCATION_DESCRIPTOR_WITH_ADDITIONAL_DATA} set to true.
+ */
+
+@Data
+class HiveLocationDescriptor {
+  public static final String HIVE_DATASET_COPY_ADDITIONAL_PATHS_ENABLED= HiveDatasetFinder.HIVE_DATASET_PREFIX + ".copy.additional.paths.enabled";
+  
+  protected final Path location;
+  protected final InputFormat<?, ?> inputFormat;
+  protected final FileSystem fileSystem;
+  protected final Properties properties;
+
+  public Set<Path> getPaths() throws IOException {
+    Set<Path> result = HiveUtils.getPaths(this.inputFormat, this.location);
```

Comment:
This is unnecessary when we are just copying file recursively. It could be moved to the else condition below.